embedded/store/immustore_test.go

```diff
@@ -1016,6 +1016,29 @@ func TestImmudbStoreRWTransactions(t *testing.T) {
 		require.NoError(t, err)
 		require.NotNil(t, valRef)
 		require.True(t, valRef.KVMetadata().Deleted())
+		require.NotNil(t, valRef.KVMetadata())
+		require.False(t, valRef.KVMetadata().Expirable())
+	})
+
+	t.Run("expired keys should not be reachable", func(t *testing.T) {
+		now := time.Now()
+
+		tx, err := immuStore.NewTx()
+		require.NoError(t, err)
+
+		err = tx.Set([]byte("expirableKey"), NewKVMetadata().ExpiresAt(&now), []byte("expirableValue"))
```

Comment:
This could also check another key with expiration date but that has not yet expired